packages/mobile/src/exchange/reducer.ts

```diff
@@ -1,28 +1,89 @@
 import { Actions, ActionTypes } from 'src/exchange/actions'
+import { getRehydratePayload, REHYDRATE, RehydrateAction } from 'src/redux/persist-helper'
 import { RootState } from 'src/redux/reducers'
 
+const MAX_HISTORY_RETENTION = 180 * 24 * 3600 // 180 days
+
 export interface ExchangeRatePair {
   goldMaker: string // number of dollarTokens received for one goldToken
   dollarMaker: string // number of goldTokens received for one dollarToken
 }
 
+export interface ExchangeRate {
+  exchangeRate: string
+  timestamp: number
+}
+
 export interface State {
   exchangeRatePair: ExchangeRatePair | null
+  history: {
+    isLoading: boolean
+    celoGoldExchangeRates: ExchangeRate[]
+    lastTimeUpdated: number
+  }
 }
 
 const initialState = {
   exchangeRatePair: null,
+  history: {
+    isLoading: false,
+    celoGoldExchangeRates: [],
+    lastTimeUpdated: 0,
+  },
 }
 
 export const exchangeRatePairSelector = (state: RootState) => state.exchange.exchangeRatePair
+export const exchangeHistorySelector = (state: RootState) => state.exchange.history
 
-export const reducer = (state: State | undefined = initialState, action: ActionTypes): State => {
+export const historyReducer = (
+  state: State['history'] | undefined = initialState.history,
+  action: ActionTypes
+): State['history'] => {
   switch (action.type) {
+    case Actions.UPDATE_CELO_GOLD_EXCHANGE_RATE_HISTORY:
```

Comment:
why the switch cases here when this reducer is only used for the `UPDATE_CELO_GOLD_EXCHANGE_RATE_HISTORY` case below?

Reply:
This is a good catch!

Reply:
do you still need to fix this?

Reply:
Not anymore, cause `UPDATE_CELO_GOLD_EXCHANGE_RATE_HISTORY` is the only event now related to `history`